core/src/main/scala/akka/kafka/scaladsl/DiscoverySupport.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.discovery.Discovery
+import akka.kafka.{ConsumerSettings, ProducerSettings}
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Scala API.
+ *
+ * Reads Kafka bootstrap servers from configured sources via [[akka.discovery.Discovery]] configuration.
+ */
+object DiscoverySupport {
+
+  /**
+   * Use Akka Discovery to read the addresses for `serviceName` within `lookupTimeout`.
+   */
+  private def bootstrapServers(
+      serviceName: String,
+      lookupTimeout: FiniteDuration
+  )(implicit system: ActorSystem): Future[String] = {
+    import system.dispatcher
+    val discovery = Discovery(system).discovery
```

Comment:
do you want to support looking up a configured discovery mechanism or just support the default one?

Reply:
Isn't that configured ActorSystem-wide?

Reply:
I think it would be useful to support overriding the default (ActorSystem-wide) setting with an alternative for this specific purpose. Cluster Bootstrap has this for example:

https://github.com/akka/akka-management/blob/d0e1baa3df4985089da7a031fc3c4abfb7013a4b/cluster-bootstrap/src/main/resources/reference.conf#L55-L61
https://github.com/akka/akka-management/blob/d0e1baa3df4985089da7a031fc3c4abfb7013a4b/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala#L42-L53
https://github.com/akka/akka-management/blob/d0e1baa3df4985089da7a031fc3c4abfb7013a4b/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrapSettings.scala#L62

This came up today, with a Lagom user who wanted to use a different discovery method just for Kafka, which Lagom does not currently support. In the end, I don't think they really needed this, and found another suitable approach. I don't think it's critical, but if someone does turn out to need it, they'd have to reimplement most of this class, just to change that one thing.

Reply:
I'm not sure adding another level of indirection is so important in this case. The user can easily create their own implementation of just this `DiscoverySupport` and use it with `enrichAsync` instead.